src/deadline_worker_agent/sessions/actions/run_attachment_download.py

```diff
@@ -259,6 +281,7 @@ def start(
             # successfully launched VFS
             return
 
+        dynamic_mapping_rules.update(generated_path_mapping)
```

Comment:
nit: this could be up in the new `if` block to keep it closer. Also a no-op if the branch is skipped